File: 1.py
# # 1. Kasus Absensi Kelas (Linear Search) 
# def cari_absensi(daftar, target):
#     for i in range(len(daftar)):
#         if daftar[i] == target:
#             return i
#     return -1

# hasil1 = cari_absensi(["Izzan", "Susilo", "Jefri", "Nichol"], "Susilo")
# print(f"1. Absensi 'Susilo': {'posisi ke-' + str(hasil1 + 1) if hasil1 != -1 else 'Data tidak ditemukan'}")

# 2. Kasus Lost and Found (Linear Search) 
# def cari_barang(tumpukan, target):
#     for i in range(len(tumpukan)):
#         if tumpukan[i] == target:
#             return i
#     return -1

# hasil2 = cari_barang(["Dompet", "Kunci Motor", "Buku"], "Kunci Motor")
# print(f"2. Lost & Found 'Kunci Motor': {'posisi ke-' + str(hasil2 + 1) if hasil2 != -1 else 'Data tidak ditemukan'}")

# 3. Kasus Buku Perpustakaan (Binary Search)
# def cari_buku(daftar_seri, target):
#     rendah = 0
#     tinggi = len(daftar_seri) - 1
#     while rendah <= tinggi:
#         tengah = (rendah + tinggi) // 2
#         if daftar_seri[tengah] == target:
#             return tengah
#         elif daftar_seri[tengah] < target:
#             rendah = tengah + 1
#         else:
#             tinggi = tengah - 1
#     return -1

# hasil3 = cari_buku([101, 102, 105, 110], 105)
# print(f"3. Buku Seri 105: {'posisi ke-' + str(hasil3 + 1) if hasil3 != -1 else 'Data tidak ditemukan'}")

# 4. Kasus Ranking Tryout (Binary Search - Menurun)
# def cari_ranking(skor_list, target):
#     rendah = 0
#     tinggi = len(skor_list) - 1
#     while rendah <= tinggi:
#         tengah = (rendah + tinggi) // 2
#         if skor_list[tengah] == target:
#             return tengah
#         elif skor_list[tengah] < target:
#             tinggi = tengah - 1
#         else:
#             rendah = tengah + 1
#     return -1

# hasil4 = cari_ranking([990, 950, 890, 800], 890)
# print(f"4. Ranking Skor 890: {'posisi ke-' + str(hasil4 + 1) if hasil4 != -1 else 'Data tidak ditemukan'}")

# 5. Kasus Kamus Istilah (Binary Search)
# def cari_kamus(kamus, target):
#     rendah = 0
#     tinggi = len(kamus) - 1
#     while rendah <= tinggi:
#         tengah = (rendah + tinggi) // 2
#         if kamus[tengah] == target:
#             return tengah
#         elif kamus[tengah] < target:
#             rendah = tengah + 1
#         else:
#             tinggi = tengah - 1
#     return -1

# hasil5 = cari_kamus(["Algorithm", "Binary", "Data"], "Algorithm")
# print(f"5. Kamus 'Algorithm': {'posisi ke-' + str(hasil5 + 1) if hasil5 != -1 else 'Data tidak ditemukan'}")

File: 1.php
<?php
// Fungsi-fungsi Pencarian
// function cariAbsensi($daftar, $target) {
//     foreach ($daftar as $i => $v) if ($v === $target) return $i;
//     return -1;
// }

// $hasil1 = cariAbsensi(["Budi", "Siti", "Rian", "Andi"], "Rian");
// echo "1. Absensi 'Rian': " . ($hasil1 !== -1 ? "posisi ke-" . ($hasil1 + 1) : "Data tidak ditemukan") . "\n";



// function cariBarang($tumpukan, $target) {
//     foreach ($tumpukan as $i => $v) if ($v === $target) return $i;
//     return -1;
// }

// $hasil2 = cariBarang(["Dompet", "Kunci Motor", "Buku"], "Kunci Motor");
// echo "2. Lost & Found 'Kunci Motor': " . ($hasil2 !== -1 ? "posisi ke-" . ($hasil2 + 1) : "Data tidak ditemukan") . "\n";



// function cariBuku($daftarSeri, $target) {
//     $low = 0; $high = count($daftarSeri) - 1;
//     while ($low <= $high) {
//         $mid = floor(($low + $high) / 2);
//         if ($daftarSeri[$mid] == $target) return $mid;
//         if ($daftarSeri[$mid] < $target) $low = $mid + 1;
//         else $high = $mid - 1;
//     }
//     return -1;
// }

// $hasil3 = cariBuku([101, 102, 105, 110], 105);
// echo "3. Buku Seri 105: " . ($hasil3 !== -1 ? "posisi ke-" . ($hasil3 + 1) : "Data tidak ditemukan") . "\n";



// function cariRanking($skorList, $target) {
//     $low = 0; $high = count($skorList) - 1;
//     while ($low <= $high) {
//         $mid = floor(($low + $high) / 2);
//         if ($skorList[$mid] == $target) return $mid;
//         if ($skorList[$mid] < $target) $high = $mid - 1;
//         else $low = $mid + 1;
//     }
//     return -1;
// }

// $hasil4 = cariRanking([990, 950, 890, 800], 890);
// echo "4. Ranking Skor 890: " . ($hasil4 !== -1 ? "posisi ke-" . ($hasil4 + 1) : "Data tidak ditemukan") . "\n";



// function cariKamus($kamus, $target) {
//     $low = 0; $high = count($kamus) - 1;
//     while ($low <= $high) {
//         $mid = floor(($low + $high) / 2);
//         if ($kamus[$mid] === $target) return $mid;
//         if ($kamus[$mid] < $target) $low = $mid + 1;
//         else $high = $mid - 1;
//     }
//     return -1;
// }

// $hasil5 = cariKamus(["Algorithm", "Binary", "Data"], "Algorithm");
// echo "5. Kamus 'Algorithm': " . ($hasil5 !== -1 ? "posisi ke-" . ($hasil5 + 1) : "Data tidak ditemukan") . "\n";
?>